File: Backend/connecthub/src/main/java/com/example/connecthub/service/PostService.java
package com.example.connecthub.service;

import com.example.connecthub.dto.request.CreatePostRequest;
import com.example.connecthub.dto.request.UpdatePostRequest;
import com.example.connecthub.dto.response.PostResponse;

import java.util.List;


public interface PostService {

    PostResponse createPost(
            String email,
            CreatePostRequest request
    );

    List<PostResponse> getAllPosts();

    List<PostResponse> getPostsByCommunity(Long communityId);

    PostResponse getPostById(Long id);

    PostResponse updatePost(
            Long id,
            String email,
            UpdatePostRequest request
    );

    void deletePost(
            Long id,
            String email
    );
}

File: frontend/src/services/postService.js
import client from '../api/client';

export const postService = {
  getAllPosts: async () => {
    const response = await client.get('/posts');
    return response.data; // List<PostResponse>
  },

  getPostById: async (id) => {
    const response = await client.get(`/posts/${id}`);
    return response.data; // PostResponse
  },

  getPostsByCommunity: async (communityId) => {
    const response = await client.get(`/posts/community/${communityId}`);
    return response.data; // List<PostResponse>
  },

  createPost: async (postData) => {
    // { title, content, imageUrl, communityId }
    const response = await client.post('/posts', postData);
    return response.data; // PostResponse
  },

  updatePost: async (id, updateData) => {
    // { title, content, imageUrl }
    const response = await client.put(`/posts/${id}`, updateData);
    return response.data; // PostResponse
  },

  deletePost: async (id) => {
    await client.delete(`/posts/${id}`);
  },
};
